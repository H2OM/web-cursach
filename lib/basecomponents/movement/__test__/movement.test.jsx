import Movement from "../movement";
import { render, screen } from "@testing-library/react";


it('movement render test', async ()=>{
    const blocks = ['first', 'second', 'third'];

    const childrens = blocks.map(each=>{
        return <div key={each}>{each}</div>
    });
    const {container} = render(<Movement children={childrens}/>);

    const navs = screen.getAllByRole('navRadio');
    
    expect(navs.length).toBe(3);

    blocks.forEach(each=>{
        expect(screen.getByText(each)).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
    
});