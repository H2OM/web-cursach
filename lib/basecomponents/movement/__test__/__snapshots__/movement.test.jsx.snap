// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movement render test 1`] = `
<div>
  <div
    class="Movement__wrap"
  >
    <div
      class="Movement__slider"
      style="transform: translateX(-0%);"
    >
      <div>
        first
      </div>
      <div>
        second
      </div>
      <div>
        third
      </div>
    </div>
    <nav
      class="Movement__nav"
    >
      <div
        class="Movement__nav__point Movement__nav__point_select"
        role="navRadio"
      />
      <div
        class="Movement__nav__point "
        role="navRadio"
      />
      <div
        class="Movement__nav__point "
        role="navRadio"
      />
    </nav>
  </div>
</div>
`;
